src/DrawerNavigation.re

```diff
@@ -0,0 +1,54 @@
+type navigation('a) = {navigate: 'a => unit};
+
+[@bs.deriving abstract]
+type drawerOptions = {
+  [@bs.optional]
+  activeTintColor: string,
+};
+
+[@bs.deriving abstract]
+type screenOptions = {
+  [@bs.optional]
+  title: string,
+};
+
+module type TabConfig = {
+  type item;
+  let items: list(item);
+  let drawerOptions: drawerOptions;
+  let getItem:
+    item => (string, unit => ReasonReact.reactElement, screenOptions);
+};
+
+module Create = (Config: TabConfig) => {
+  [@bs.deriving abstract]
+  type navigatorConfig = {initialRouteName: string};
+
+  [@bs.deriving abstract]
+  type routeConfig = {
+    screen: unit => ReasonReact.reactElement,
+    navigationOptions: screenOptions,
+  };
+
+  let items =
+    Config.items
+    |> List.map(tab => {
+         let (tabname, screen, screenOptionsConfig) = Config.getItem(tab);
+         (
+           tabname,
+           routeConfig(~screen, ~navigationOptions=screenOptionsConfig),
```

Comment:
Here we can wrap screen in a (unit => screen) function if needed. I would simplify the user-facing API as much as possible.